src/Component/Generator.php

```diff
@@ -0,0 +1,182 @@
+
+            $logs = array('channels'=>array(), 'xml'=>array(),'failed_providers'=>array());
+
+            foreach($channels as $channelKey => $channelInfo) {
+                $providers = $this->getProviders($channelInfo["priority"] ?? []);
+                foreach($this->listDate as $date) {
+                    $cacheKey = sprintf("%s_%s.xml", $channelKey, $date);
+                    if(!isset($logs["channels"][$date][$channelKey])){
+                        $logs["channels"][$date][$channelKey] = [
+                            'success' => false,
+                            'provider' => null,
+                            'cache'=> false,
+                            'failed_providers' => [],
+                        ];
+                    }
+                    Logger::log(sprintf("\e[95m[EPG GRAB] \e[39m%s : %s", $channelKey, $date));
+
+                    if ($this->cache->has($cacheKey)){
+                        Logger::log(" | \e[33mOK \e[39m- From Cache ".chr(10));
+                        continue;
+                    }
+                    $channelFound = false;
+                    foreach ($providers as $provider) {
+                        $old_zone = date_default_timezone_get();
+                        $channel = $provider->constructEPG($channelKey, $date);
+                        date_default_timezone_set($old_zone);
+                        if($channel !== false){
+                            $channelFound = true;
+                            $logs['channels'][$date][$channelKey] = [
+                                'success' => true,
+                                'provider' => get_class($provider),
+                                'cache'=> false,
+                            ];
+                            $this->cache->store($cacheKey, $this->formatter->formatChannel($channel, $provider));
+                            Logger::log(" | \e[32mOK\e[39m - ".Utils::extractProviderName($provider).chr(10));
+                            break ;
+                        }
+
+                        $logs['channels'][$date][$channelKey]['failed_providers'][] = get_class($provider);
+                        $logs['failed_providers'][get_class($provider)] = true;
+                    }
+
+                    if(!$channelFound) {
+                        if ($this->createEpgIfNotFound) {
+                            $this->cache->store($cacheKey, $this->formatter->formatChannel(new DummyChannel($channelKey, $date), null));
+                        }
+                        Logger::log(" | \e[31mHS\e[39m".chr(10));
+                    }
+                }
+            }
+            Logger::log("\e[95m[EPG GRAB] \e[39mRécupération du guide des programmes terminée...\n");
+            $log_path = 'logs/logs'.date('YmdHis').'.json';
+            Logger::log("\e[36m[LOGS] \e[39m Export des logs vers $log_path\n");
+
+            //debug information
+            file_put_contents($log_path,json_encode($logs));
+        }
+    }
```

Comment:
On perd certains paramètres comme "force_todays_grab" qui forçait la récupération pour l'EPG du jour.

Reply:
Alors le today_grab est géré diffèremment.
C'est gerer directement dans le constructeur, si today_grab est a vrai, alors la date de début est aujourd'hui.